src/org/labkey/test/pages/dataset/EditDatasetDefinitionPage.java

```diff
@@ -0,0 +1,282 @@
+        elementCache().participantsOnlyRadioBtn.set(checked);
+        return this;
+    }
+
+    public boolean isDemographicsData()
+    {
+        return  elementCache().participantsOnlyRadioBtn.get();
+    }
+
+    public String getAdditionalKeyColDataField()
+    {
+        if (!elementCache().keyFieldSelect.isEnabled())
+            return null;
+        else
+            return elementCache().keyFieldSelect.getValue();
+    }
+
+    public EditDatasetDefinitionPage setAdditionalKeyColDataField(String field)
+    {
+        setAdditionalKeyColumnType(LookupAdditionalKeyColType.MANAGEDFIELD);
+        elementCache().keyFieldSelect.select(field);
+        return this;
+    }
+
+    public EditDatasetDefinitionPage setAdditionalKeyColManagedField(String field)
+    {
+        setAdditionalKeyColumnType(LookupAdditionalKeyColType.MANAGEDFIELD);
+        elementCache().keyFieldSelect.select(field);
+        elementCache().keyPropertyManagedBox.check();
+        return this;
+    }
+
+    public EditDatasetDefinitionPage setAdditionalKeyColumnType(LookupAdditionalKeyColType type)
+    {
+        expandPropertiesPanel();
+        new RadioButton(elementCache().dataRowRadioBtn(type.getIndex()).findElement(getDriver())).check();
+        return this;
+    }
+
+    public boolean isAdditionalKeyManagedEnabled()
+    {
+        expandPropertiesPanel();
+        return elementCache().additionalKeyFieldRadioBtn.isChecked() &&
+                elementCache().keyFieldSelect.hasValue() &&
+                elementCache().keyPropertyManagedBox.isChecked();
+    }
+
+    public boolean isAdditionalKeyDataFieldEnabled()
+    {
+        return  elementCache().keyFieldSelect.isEnabled();
+    }
+
+    public boolean isAdditionalFieldNoneEnabled()
+    {
+        return elementCache().additionalKeyFieldRadioBtn.isChecked() &&
+                !elementCache().keyFieldSelect.hasValue();
+    }
+
+    // get/select additional key field
+    // get/set 'let server manage fields to make entries unique' checkbox
+
+    public EditDatasetDefinitionPage setShowInOverview(boolean checked)
+    {
+        expandPropertiesPanel();
+        return openAdvancedDatasetSettings()
+                .setShowInOverview(checked)
+                .clickApply();
+    }
+
+    public EditDatasetDefinitionPage saveExpectFail(String expectedError)
+    {
+        List<String>  errors = clickSaveExpectingErrors();
+        assertThat("Errors on the page should include this expected one", errors, hasItem(expectedError));
+        return this;
+    }
+
+    public DatasetPropertiesPage clickSave()
+    {
+        getWrapper().doAndWaitForPageToLoad(()-> elementCache().saveButton.click());
+        return new DatasetPropertiesPage(getDriver());
+    }
+
+    public enum LookupAdditionalKeyColType
+    {
+        NONE("Participants only (demographic data)", 0),
+        DATAFIELD("Participants and visits", 1),
+        MANAGEDFIELD("Participants, visits, and additional key field", 2);
```

Comment:
The name of this enum and these values is very misleading at this point given the way this UI was refactored. I would suggest that we change this enum to be DataRowUniquenessType and the values to be DEMOGRPHIC, PTID_TIMEPOINT, ADDITIONAL_KEY

Reply:
@labkey-chrisj @labkey-sweta FYI, I am making this change now and will have it checked in shortly.

Reply:
Thanks.  It was confusing before and maybe became more confusing after the refactor.